plugin.video.itvrs/resources/lib/plugintools.py

```diff
@@ -0,0 +1,497 @@
+        itemurl = url
+        listitem.setProperty('IsPlayable', 'true')
+        xbmcplugin.addDirectoryItem( handle=int(sys.argv[1]), url=itemurl, listitem=listitem, isFolder=folder)
+    elif isPlayable:
+        listitem.setProperty("Video", "true")
+        listitem.setProperty('IsPlayable', 'true')
+        itemurl = '%s?action=%s&title=%s&url=%s&thumbnail=%s&plot=%s&extra=%s&page=%s' % ( sys.argv[ 0 ] , action , urllib.quote_plus( title ) , urllib.quote_plus(url) , urllib.quote_plus( thumbnail ) , urllib.quote_plus( plot ) , urllib.quote_plus( extra ) , urllib.quote_plus( page ))
+        xbmcplugin.addDirectoryItem( handle=int(sys.argv[1]), url=itemurl, listitem=listitem, isFolder=folder)
+    else:
+        itemurl = '%s?action=%s&title=%s&url=%s&thumbnail=%s&plot=%s&extra=%s&page=%s' % ( sys.argv[ 0 ] , action , urllib.quote_plus( title ) , urllib.quote_plus(url) , urllib.quote_plus( thumbnail ) , urllib.quote_plus( plot ) , urllib.quote_plus( extra ) , urllib.quote_plus( page ))
+        xbmcplugin.addDirectoryItem( handle=int(sys.argv[1]), url=itemurl, listitem=listitem, isFolder=folder)
+
+def close_item_list():
+    _log("close_item_list")
+
+    xbmcplugin.endOfDirectory(handle=int(sys.argv[1]), succeeded=True)
+
+def play_resolved_url(url):
+    _log("play_resolved_url ["+url+"]")
+
+    listitem = xbmcgui.ListItem(path=url)
+    listitem.setProperty('IsPlayable', 'true')
+    return xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, listitem)
+
+def direct_play(url):
+    _log("direct_play ["+url+"]")
+
+    title = ""
+
+    try:
+        xlistitem = xbmcgui.ListItem( title, path=url)
+    except:
+        xlistitem = xbmcgui.ListItem( title, )
+    xlistitem.setArt({"thumb":"DefaultVideo.png","icon":"DefaultVideo.png"})
+    xlistitem.setInfo( "video", { "Title": title } )
+
+    playlist = xbmc.PlayList( xbmc.PLAYLIST_VIDEO )
+    playlist.clear()
+    playlist.add( url, xlistitem )
+
+    player_type = xbmc.PLAYER_CORE_AUTO
+    xbmcPlayer = xbmc.Player( player_type )
+    xbmcPlayer.play(playlist)
+
+def show_picture(url):
+
+    local_folder = os.path.join(get_data_path(),"images")
+    if not os.path.exists(local_folder):
+        try:
+            os.mkdir(local_folder)
+        except:
+            pass
+    local_file = os.path.join(local_folder,"temp.jpg")
+
+    # Download picture
+    urllib.urlretrieve(url, local_file)
+    
+    # Show picture
+    xbmc.executebuiltin( "SlideShow("+local_folder+")" )
+
+def get_temp_path():
+    _log("get_temp_path")
+
+    dev = xbmc.translatePath( "special://temp/" )
+    _log("get_temp_path ->'"+str(dev)+"'")
+
+    return dev
+
+def get_runtime_path():
+    _log("get_runtime_path")
+
+    dev = xbmc.translatePath( __settings__.getAddonInfo('Path') )
+    _log("get_runtime_path ->'"+str(dev)+"'")
+
+    return dev
+
+def get_data_path():
+    _log("get_data_path")
+
+    dev = xbmc.translatePath( __settings__.getAddonInfo('Profile') )
+    
+    # Parche para XBMC4XBOX
```

Comment:
xbmc4xbox? :)
This looks too old. Please check the python3 compatibility it will certainly not work.

Reply:
I did check the python3 compatibility and it was fine.